### PR TITLE
Add select2-bootstrap to greenwich theme

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Adds bootstrap-3 styling to select2 elements.

Before
----------------------------------------
image

After
----------------------------------------
image

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
nice!

**Comment by colemanw** [comment]
@totten unfortunately the `composer-compile` step crashes. Only workaround I've found so far is to manually rename the file `bower_components/select2/select2-bootstrap.css` and give it an **`.scss`** extension.

File: ext/greenwich/composer.compile.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "php-method": "\\Civi\\Compile\\Scss::build",
       "watch-files": ["scss"],
       "scss-files": {"scss/main.scss": "dist/bootstrap3.css"},
-      "scss-includes": ["scss", "extern/bootstrap3/assets/stylesheets"]
+      "scss-includes": ["scss", "extern/bootstrap3/assets/stylesheets", "../../bower_components/select2"]
```
Comment (seamuslee001) [inline ext/greenwich/composer.compile.json:8]: @colemanw why is this needed is there an issue with the import if its not there?

**Comment by seamuslee001** [comment]
@colemanw does it make more sense to include something like https://github.com/civicrm/civicrm-core/blob/master/ext/greenwich/greenwich.php#L55 ?

**Comment by colemanw** [comment]
> @colemanw does it make more sense to include something like https://github.com/civicrm/civicrm-core/blob/master/ext/greenwich/greenwich.php#L55 ?

Then I think it wouldn't get the `#bootstrap-theme` namespace prefixing.

**Comment by totten** [comment]
> unfortunately the composer-compile step crashes. Only workaround I've found so far is to manually rename the file bower_components/select2/select2-bootstrap.css and give it an .scss extension.

I [didn't realize](https://lab.civicrm.org/dev/user-interface/-/issues/27#note_43972) that `select2-bootstrap.css` was bundled with `select2`.

OK, so for completeness, the error that appears to me is:

```
Fatal error: Uncaught ScssPhp\ScssPhp\Exception\ParserException:
Error : `border-radius: 0 2px 2px 0;` isn't allowed in plain CSS (assign): failed at `}`
../../bower_components/select2/select2-bootstrap.css on line 9, at column 1
in /home/totten/bknix/build/dmaster/web/sites/all/modules/civicrm/vendor/scssphp/scssphp/src/Parser.php:150
```

Which seems to indicate that (1) ScssPhp does allow importing `*.css` files but (2) it treats them differently than importing `*.scss`. This particular error doesn't mean much to me, but it's understandable that different file-extensions would have different import-semantics.

How's this as a fairly straight interpretation of your local work-around:

https://github.com/totten/civicrm-core/commit/89bef740fa72ada7851127cf980b7ab5c5f9b6e9

**Comment by colemanw** [comment]
Yes that's the error message I got too. I tried deleting the "offending" line in the css file and then got the same error for many more lines in the file. To me it looks like a bug in the php scss compiler, but whatever, this workaround works for now.

**Comment by totten** [comment]
OK, cool.

@colemanw  I have a slightly better form of the same patch now - it's a bit more intuitive to manage the 'watch' list, and I expect it to run better on Windows.  It needs an update version of compile-plugin v0.8. Changes pushed onto the PR.

**Comment by colemanw** [comment]
@totten I tested it and it works well.
